File: kirun-java/src/main/java/com/fincity/nocode/kirun/engine/function/Function.java
package com.fincity.nocode.kirun.engine.function;

import java.util.List;
import java.util.Map;

import com.fincity.nocode.kirun.engine.json.schema.Schema;
import com.fincity.nocode.kirun.engine.model.Event;
import com.fincity.nocode.kirun.engine.model.FunctionOutput;
import com.fincity.nocode.kirun.engine.model.FunctionSignature;
import com.fincity.nocode.kirun.engine.runtime.FunctionExecutionParameters;

public interface Function {

	public FunctionSignature getSignature();
	
	public Map<String, Event> getProbableEventSignature(Map<String, List<Schema>> probableParameters);

	public FunctionOutput execute(FunctionExecutionParameters context);
}

File: kirun-js/src/engine/function/Function.ts
import { Schema } from '../json/schema/Schema';
import { Event } from '../model/Event';
import { FunctionOutput } from '../model/FunctionOutput';
import { FunctionSignature } from '../model/FunctionSignature';
import { FunctionExecutionParameters } from '../runtime/FunctionExecutionParameters';

export interface Function {
    getSignature(): FunctionSignature;

    getProbableEventSignature(probableParameters: Map<string, Schema[]>): Map<string, Event>;

    execute(context: FunctionExecutionParameters): FunctionOutput;
}
